lib/post_creator.rb

```diff
@@ -136,6 +136,14 @@ def valid?
       return false unless skip_validations? || validate_child(topic_creator)
     else
       @topic = Topic.find_by(id: @opts[:topic_id])
+
+      if @topic.present?
```

Comment:
Not a big deal, but you could just `&&` there 2 conditions to avoid a nested if.